ex02.ex

```diff
@@ -53,12 +53,13 @@ defmodule Ex02 do
   # if the first two elements of a list are equal                #
   ################################################################
 
-  first2equal = your_anonymous_function([])
+  first2equal = fn (list) ->
```

Comment:
## -1

So I've spent at least 20 minutes trying to grade this particular answer.

I was hoping to see something like this:

~~~ elixir
  first2equal = fn
                  [ a, a | _ ]              -> true
                  other when is_list(other) -> false
                end
~~~

That uses the pattern match to check the first two elements for equality.

Your solution would be perfectly fine in general, but it doesn't really exploit pattern matching. So I was going to deduct 2 or 3.

But, it has the benefit that it doesn't need the ugly `is_list` checks.

But... if could use pattern matching better than it does, and have something like

~~~ elixir
  first2equal = fn [ a, b | _ ]  -> a == b end
~~~

So in the end, I decided to drop just one for not doing the pattern match in the function head.